client/src/components/geneExpression/index.tsx

```diff
@@ -53,39 +57,68 @@ class GeneExpression extends React.Component<{}, State> {
   }
 
   // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types --- FIXME: disabled temporarily on migrate to TS.
-  renderGeneSets = () => {
+  renderGeneSets = (getMarkerGeneSets: boolean) => {
     const sets = [];
     // @ts-expect-error ts-migrate(2339) FIXME: Property 'genesets' does not exist on type 'Readon... Remove this comment to see the full error message
-    const { genesets } = this.props;
+    const { genesets, isCellGuideCxg } = this.props;
     const { geneIds, geneNames } = this.state;
 
     for (const [name, geneset] of genesets) {
-      const genesetIds = [];
-      const genesetNames = [];
-
-      // find ensembl IDs for each gene in the geneset
-      for (const gene of geneset.genes) {
-        let geneId;
-        if (geneIds) {
-          geneId = geneIds[geneNames.indexOf(gene[0])];
-        } else {
-          geneId = "";
+      /**
+       * This conditional checks if the current geneset should be displayed based on the following criteria:
+       * 1. If the geneset name does not include MARKER_GENE_SUFFIX_IDENTIFIER, it's not a marker geneset and we're not
+       * looking for marker genesets,and it's a CellGuide CXG dataset, then it should be displayed.
+       * This yields non-marker genesets in CellGuide Explorer instances.
+       *
+       * 2. If the geneset name includes MARKER_GENE_SUFFIX_IDENTIFIER, it's a marker geneset and we're looking for marker
+       * genesets, and it's a CellGuide CXG dataset, then it should be displayed.
+       * This yields marker genesets in CellGuide Explorer instances.
+       *
+       * 3. If it's not a CellGuide CXG dataset, then there's no filtering or grouping based on geneset type,
+       * so it should be displayed.
+       * This is the behavior for genesets in non-CellGuide Explorer instances.
+       */
+      if (
+        (!name.includes(MARKER_GENE_SUFFIX_IDENTIFIER) &&
+          !getMarkerGeneSets &&
+          isCellGuideCxg) ||
+        (name.includes(MARKER_GENE_SUFFIX_IDENTIFIER) &&
+          getMarkerGeneSets &&
+          isCellGuideCxg) ||
+        !isCellGuideCxg // if not a CellGuide CXG, don't do any filtering or grouping
```

Comment:
The comments are helpful, but I wonder if we'd get better readability and DRY by breaking down the conditions into their own variables

E.g.,

```ts
const isMarkerGeneSet = name.includes(MARKER_GENE_SUFFIX_IDENTIFIER);

const condition1 = !isMarkerGeneSet && !getMarkerGeneSets && isCellGuideCxg;
const condition2 = isMarkerGeneSet && getMarkerGeneSets && isCellGuideCxg; 
const condition3 = !isCellGuideCxg; 

if (condition1 || condition2 || condition3) {
  ...
}
```